ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -0,0 +1,355 @@
+                    peer_groups[self._peer_group_mpls_overlay_peers]["default_route_target"] = {"only": True}
+
+            if self._overlay_evpn_vxlan is True:
+                if self._evpn_role == "server" or self._mpls_overlay_role == "server":
+                    peer_groups[self._peer_group_evpn_overlay_peers]["default_route_target"] = {"only": True}
+
+        address_family_rtc["peer_groups"] = peer_groups
+
+        return address_family_rtc
+
+    def _address_family_vpn_ipvx(self, version: int) -> dict | None:
+        if version not in [4, 6]:
+            raise AristaAvdError("_address_family_vpn_ipvx should be called with version 4 or 6 only")
+
+        if (version == 4 and self._overlay_vpn_ipv4 is not True) or (version == 6 and self._overlay_vpn_ipv6 is not True):
+            return None
+
+        address_family_vpn_ipvx = {}
+
+        if self._overlay_ler is True or self._overlay_ipvpn_gateway is True:
+            address_family_vpn_ipvx["neighbor_default_encapsulation_mpls_next_hop_self"] = {"source_interface": "Loopback0"}
+
+        peer_groups = {}
+
+        if self._overlay_ipvpn_gateway is True:
+            peer_groups[self._peer_group_ipvpn_gateway_peers] = {"activate": True}
+
+        if self._overlay_routing_protocol == "ibgp":
+            if self._overlay_mpls is True:
+                peer_groups[self._peer_group_mpls_overlay_peers] = {"activate": True}
+
+            if self._peer_group_rr_overlay_peers is not None and self._mpls_overlay_role == "server":
+                peer_groups[self._peer_group_rr_overlay_peers] = {"activate": True}
+
+        if peer_groups:
+            address_family_vpn_ipvx["peer_groups"] = peer_groups
+
+        if self._overlay_dpath is True:
+            address_family_vpn_ipvx["domain_identifier"] = get(self._hostvars, "switch.ipvpn_gateway.ipvpn_domain_id", required=True)
+
+        return address_family_vpn_ipvx
+
+    def _create_neighbor(self, name: str, peer_group: str, remote_as: str = None) -> dict:
+        """ """
+        neighbor = {"peer_group": peer_group, "description": name}
+
+        if self._overlay_routing_protocol == "ebgp":
+            if remote_as is None:
+                raise AristaAvdError("Configuring eBGP neighor without a remote_as")
+
+            neighbor["remote_as"] = remote_as
+
+        return neighbor
+
+    def _neighbors(self) -> dict | None:
+        """ """
+        neighbors = {}
+
+        if self._overlay_routing_protocol == "ebgp":
+            for route_server, data in natural_sort(self._evpn_route_servers.items()):
+                neighbor = self._create_neighbor(route_server, self._peer_group_evpn_overlay_peers, remote_as=data["bgp_as"])
+                if self._evpn_prevent_readvertise_to_server is True:
+                    neighbor["route_map_out"] = f"RM-EVPN-FILTER-AS{self._evpn_route_servers[route_server]['bgp_as']}"
+                neighbors.update({data["ip_address"]: neighbor})
```

Comment:
Why not just set `router_server_ip` once and then assign `neighbors[route_server_ip] = neighbor` ? Do you expect that the same device will be caught multiple times? (If so it will be very hard to predict the outcome with the current updates).

Reply:
Indeed I don't need to use updates here